container_opts_unix.go

```diff
@@ -188,21 +187,15 @@ func withRemappedSnapshotBase(id string, i Image, uid, gid uint32, readonly bool
 }
 
 func remapRootFS(mounts []mount.Mount, uid, gid uint32) error {
-	root, err := ioutil.TempDir("", "ctd-remap")
-	if err != nil {
-		return err
-	}
-	defer os.Remove(root)
-	for _, m := range mounts {
-		if err := m.Mount(root); err != nil {
-			return err
-		}
-	}
-	err = filepath.Walk(root, incrementFS(root, uid, gid))
-	if uerr := mount.Unmount(root, 0); err == nil {
-		err = uerr
+	mountOption := []mount.MountsOption{
+		{
+			Mounts:       mounts,
+			TmpDirPrefix: "ctd-remap",
+		},
 	}
-	return err
+	return mount.VisitMountedDir(mountOption, func(roots []string) error {
```

Comment:
Maybe change the signature to `mount.Visit(mounts []Mount, parentDir, prefix string, func(rootfs []string) error)`

And let the caller do a loop over it if it needs to?

Reply:
At first I implement it as you say,  but  walkingDiff.DiffMounts need to  handle two temp dir at the same time.